src/add-card-to-project.ts

```diff
@@ -0,0 +1,50 @@
+import { Probot,Context } from 'probot';
+import { LogLevel } from './enums';
+import { log } from './utils/log-util';
+
+export function addCardToProject(probot: Probot) {
+    probot.on("issues.opened", async (context:Context['octokit']) => {
```

Comment:
The association would occur at the point the issue is labeled, not opened - this should be `issues.opened`.

Hooking on this event also gives us access to the specific label that was added - this then tells us which board we would be associating it to. You can pull `label` from `context.payload` to get that information.